vcr/issuer/interface.go

```diff
@@ -0,0 +1,43 @@
+package issuer
+
+import (
+	ssi "github.com/nuts-foundation/go-did"
+	"github.com/nuts-foundation/go-did/did"
+	"github.com/nuts-foundation/go-did/vc"
+	"github.com/nuts-foundation/nuts-node/crypto"
+	"github.com/nuts-foundation/nuts-node/vcr/credential"
+)
+
+// Publisher publishes new credentials and revocations to a channel. Used by a credential issuer.
+type Publisher interface {
+	PublishCredential(verifiableCredential vc.VerifiableCredential, public bool) error
+	PublishRevocation(revocation credential.Revocation) error
+}
+
+type keyResolver interface {
+	ResolveAssertionKey(issuerDID did.DID) (crypto.Key, error)
+}
+
+// Issuer is a role in the network for a party who issues credentials about a subject to a holder.
+type Issuer interface {
+	Issue(unsignedCredential vc.VerifiableCredential, publish, public bool) (*vc.VerifiableCredential, error)
+	Revoke(credentialID ssi.URI) error
+	CredentialResolver() StoreResolver
```

Comment:
Feels like bad design to return an internal type/object, especially a store.

"Do not call friends of friends" (do not call methods on dependencies of a type's dependencies)